test/runtests.jl

```diff
@@ -1,8 +1,20 @@
 using Diffusion
 using Test
+using StaticArrays
 
 @testset "Diffusion.jl" begin
     # Write your tests here.
-    @test absorb_boundry(1,2) == 3
+    @test absorb_boundry(SA[3,4],6) == SA[3,4]
+    @test absorb_boundry(SA[3,4],5) === nothing
+    @test absorb_boundry(SA[3,4],4) === nothing 
+    @test slope(SA[1,2],SA[3,4]) == 1
```

Comment:
If your internal functions get *used* by your public functions, often you don't have to check them directly. Indeed, there are cases where it is better not to check them, because you can think of them as an "internal implementation detail" subject to revision rather than a promise to users about the interface you'll maintain. In general, think carefully about deleting tests, which means that you generally should focus your tests on things you want to preserve long-term.

But there are many exceptions in which it's a good idea to check any, especially for anything the least bit complicated. Not exporting the function can help distinguish which things you intend to be user-supported vs internal details.

Nevertheless, you'll see all my comments are focused on your more high-level functions.